Shared/sdk/SharedUtil.Crypto.h

```diff
@@ -28,4 +30,56 @@ namespace SharedUtil
 
         return result;
     }
+
+    inline SString Aes128encode(const SString& sData, const SString& sKey, const SString& sIv)
+    {
+		SString result;
+
+		CryptoPP::byte iv[ CryptoPP::AES::BLOCKSIZE ];
+		CryptoPP::byte key[ CryptoPP::AES::DEFAULT_KEYLENGTH ];
+
+		memcpy(iv, sIv.c_str(), CryptoPP::AES::BLOCKSIZE );
+		memcpy(key, sKey.c_str(), CryptoPP::AES::DEFAULT_KEYLENGTH);
+
+		CryptoPP::AES::Encryption aesEncryption(key, CryptoPP::AES::DEFAULT_KEYLENGTH);
+		CryptoPP::CBC_Mode_ExternalCipher::Encryption cbcEncryption( aesEncryption, iv );
+
+		try
+		{
+			CryptoPP::StreamTransformationFilter stfEncryptor(cbcEncryption, new CryptoPP::StringSink( result ) );
+			stfEncryptor.Put( reinterpret_cast<const unsigned char*>( sData.c_str() ), sData.length() );
+			stfEncryptor.MessageEnd();
+		}
+		catch(const CryptoPP::Exception& exception) {
+			return exception.what();
```

Comment:
Good luck figuring out an exception was thrown in the caller function.

Reply:
Changed to "return false"

Reply:
I think it'd be desireable that the Lua function returns false in this case.

Reply:
I think you shouldn't catch this exception here, and just let the caller handle it (keep in mind the new parser only handles `std::invalid_argument`)

Reply:
Our new policy is to hard error on usage mistakes, isnt it? 
Although I'm not sure what throws here and what. If it throws for memory allocation errors as well, that should be handled differently (and most likely it does). 
I think the key is automatically padded by the algorithm, so that cant be a usage mistake. 
Maybe we should throw invalid_argument if the `iv` (in AES decode) isn't of the desired length (BLOCKSIZE), instead of resizing it.

Reply:
I was getting an exception in the first revision when CBC mode was used. Exception was thrown when an incorrect decryption key was passed. Now we use CTR mode, and this problem no longer exists. I tested different scenarios and didn't get an exception. I can assume that it could be removed. Or leave it just in case.

Reply:
> I think the key is automatically padded by the algorithm, so that cant be a usage mistake 
its not

Reply:
@Pirulax what else do you consider necessary to fix?

Reply:
Any updates?